docs/glossary.md

```diff
@@ -0,0 +1,161 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a branch that doesn't have any [branches](#branch)
+pointing to it or to any of its descendants. Unlike Git, Jujutsu keeps such
+commits around until they are explicitly abandoned.
```

Comment:
The language "such commits" is a little surprising because commits are not discussed yet in this paragraph.

Reply:
Good point, rephrased.